reactive/etcd.py

```diff
@@ -487,6 +487,20 @@ def tls_state_control():
     set_state('etcd.ssl.placed')
 
 
+@when('etcd.ssl.placed')
+@when_any('tls_client.ca.written',
+          'tls_client.server.certificate.written',
+          'tls_client.client.certificate.written')
+def tls_update():
+    ''' Handle changes to the TLS data by ensuring that the service is
+        restarted.
+    '''
+    remove_state('etcd.ssl.placed')
+    remove_state('tls_client.ca.written')
+    remove_state('tls_client.server.certificate.written')
+    remove_state('tls_client.client.certificate.written')
```

Comment:
How does this lead to a restart of the etcd service?

Reply:
Hrm.  It seems that it actually doesn't; I was going based on the `upgrade-charm` handler, where it seemed like removing the `etcd.ssl.placed` flag triggers updating the config appropriately, but looking deeper it seems like that doesn't really accomplish much.

One thing it does do is ensure the cert data is re-echoed to the `db` relations, which in practice seems to have been enough for my testing to work where it didn't without this change.  I do agree that it would be better to ensure the config is re-written and the service restarted.

Reply:
This should be addressed with the latest commit.